### ISSUE TITLE
[JENKINS-12063] Automatically update graphs when job finishes

### ISSUE THREAD

**Issue by timja-bot** [timja-org/jenkins-gh-issues-poc-11-04#8270]
<p>You are on a job page, the job is in progress, regardless of whether there is compiler warnings graph (2+ jobs run before) or no graph (only one job run before), when the job finishes, the build status updates but not the graph.</p>

<p>Trend graphs are represented by static HTML in the moment. They will be updated only if the page is refreshed (see refresh interval). </p>

---
<details><summary><i>Originally reported by <a title="Greg Moncreaff" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=moncreaff">moncreaff</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-12063" target="_blank">Automatically update graphs when job finishes</a></i></summary>
<i><ul>
<li><b>status</b>: Closed
<li><b>priority</b>: Minor
<li><b>resolution</b>: Won't Fix
<li><b>resolved</b>: 2021-05-06T21:36:56+00:00
<li><b>imported</b>: 2025-11-05
</ul></i>
</details>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=moncreaff">moncreaff</a>:</i>
<p>Is the build history not gated by refresh delay?  Should it trigger a refresh regardless of the delay when job finishes, or are these to things completely independent of each other?</p>

**Comment by timja-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=drulli">drulli</a>:</i>
<p>Ok, now I understand. I think the build history is updated via AJAX calls, the build graphs are static HTML (PNG files). They are only updated upon a refresh. I'm not sure if this could be changed easily. Anyway, that part should be changed in core so that all plug-ins will benefit from such a change. </p>
